### PR TITLE
Add `fetch_project_from_timemanager` to `TimeManagerService`

### PR DESCRIPTION
**Pull request by Pinchett-i**

### CODE REVIEW

**Review by RoelandMatthijssens — CHANGES_REQUESTED** [review]

File: services/timemanager_service.mjs [hunk 1]
```diff
@@ -1,19 +1,42 @@
...
+
+    await Promise.all(times.map(async time_registration => {
+      sum += await time_registration.billable_hours
+    }))
     return sum
   }
 
   static async time_registrations_for_project(project) {
-    return time_registrations // Fixture for now
+    let response = await this.fetch_project_from_timemanager(project)
```
[the remark this answers is not in the document]
Comment (Pinchett-i) [reply services/timemanager_service.mjs:16]: itt's the ladder , i misnamed it :+1:

File: services/timemanager_service.mjs [hunk 2]
```diff
@@ -13,11 +13,11 @@ class TimeManagerService {
   }
 
   static async time_registrations_for_project(project) {
-    let response = await this.fetch_project_from_timemanager(project)
+    let response = await this.fetch_project_time_regisitration_from_timemanager(project)
     return this.parse_response(response)
   }
 
-  static async fetch_project_from_timemanager(project) {
+  static async fetch_project_time_regisitration_from_timemanager(project) {
```
Comment (RoelandMatthijssens) [inline services/timemanager_service.mjs:20]: typo

File: services/timemanager_service.mjs [hunk 3]
```diff
@@ -1,19 +1,42 @@
...
   static async hours_for_project(project) {
     let sum = 0;
     let times = await this.time_registrations_for_project(project)
-    times.forEach(time_registration => {
-      sum += time_registration.billable_hours
-    })
+
+    await Promise.all(times.map(async time_registration => {
+      sum += await time_registration.billable_hours
+    }))
```
Comment (RoelandMatthijssens) [inline services/timemanager_service.mjs:11]: all of this async stuff can go away, calling `time_registration.billable_hours` isn't async

File: services/timemanager_service.mjs [hunk 4]
```diff
@@ -1,19 +1,42 @@
...
 
 class TimeManagerService {
   static async hours_for_project(project) {
     let sum = 0;
     let times = await this.time_registrations_for_project(project)
-    times.forEach(time_registration => {
-      sum += time_registration.billable_hours
-    })
+
+    await Promise.all(times.map(async time_registration => {
```
Comment (RoelandMatthijssens) [inline services/timemanager_service.mjs:9]: probably want to use `reduce` rather than `map`

File: services/timemanager_service.mjs [hunk 5]
```diff
@@ -1,19 +1,42 @@
...
+  static async fetch_project_time_regisitration_from_timemanager(project) {
+    return https.get(this.request_options(project), response => {
+      return this.parse_response(response)
+    })
+  }
+
+  static request_options(project) {
+    return {
+      host: config.get('timemanager.host'),
+      path: '/api/timeregistrations/query',
```
Comment (RoelandMatthijssens) [inline services/timemanager_service.mjs:29]: is this really how you want to call the endpoint?

**Review by RoelandMatthijssens — CHANGES_REQUESTED** [review]

**Review by RoelandMatthijssens — APPROVED** [review]

**Pull request merged**